instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RouteAlternativesTest.kt

```diff
@@ -113,11 +116,150 @@ class RouteAlternativesTest : BaseTest<EmptyTestActivity>(EmptyTestActivity::cla
+     */
+    @Test
+    fun additional_alternative_is_not_force_to_invoke_alternatives_observer() {
+        // Prepare with alternative routes.
+        setupMockRequestHandlers(coordinates)
+        val primaryRouteOnly = listOf(requestDirectionsRouteSync(coordinates).first())
+
+        // Play the slower alternative route.
+        runOnMainSync {
+            mapboxNavigation.historyRecorder.startRecording()
+            mockLocationReplayerRule.playRoute(primaryRouteOnly.first())
+            mapboxNavigation.startTripSession()
+        }
+
+        // Wait for enhanced locations to start and then set the routes.
+        val firstLocationIdlingResource = FirstLocationIdlingResource(mapboxNavigation)
+        firstLocationIdlingResource.firstLocationSync()
+        runOnMainSync {
+            mapboxNavigation.setRoutes(primaryRouteOnly)
+        }
+
+        // The alternative route is missed, so we expect the route to be removed.
+        val firstAlternative = RouteAlternativesIdlingResource(mapboxNavigation)
+        firstAlternative.register()
+        mapboxHistoryTestRule.stopRecordingOnCrash("alternatives failed") {
+            Espresso.onIdle()
+        }
+        firstAlternative.unregister()
+
+        assertNotNull(firstAlternative.alternatives)
+
+        val externalAlternatives = DirectionsResponse.fromJson(
+            readRawFileText(activity, R.raw.route_response_alternative_continue)
+        ).routes().also {
+            assertEquals(1, it.size)
+        }
+
+        val nextAlternativeObserver = RouteAlternativesIdlingResource(mapboxNavigation)
+
+        var setRoutes: List<DirectionsRoute> = emptyList()
+        runOnMainSync {
+            setRoutes = (mutableListOf(
+                mapboxNavigation.getRoutes().first()
+            ) + firstAlternative.alternatives!! + externalAlternatives).also {
+                assertTrue(it.size >= 2)
+            }
+
+            mapboxNavigation.setRoutes(setRoutes)
+        }
+
+        nextAlternativeObserver.register()
+        mapboxHistoryTestRule.stopRecordingOnCrash("next alternatives failed") {
+            Espresso.onIdle()
+        }
+        nextAlternativeObserver.unregister()
+
+        val countDownLatch = CountDownLatch(1)
+        runOnMainSync {
+            mapboxNavigation.historyRecorder.stopRecording {
+                logE("history path=$it", "DEBUG")
+                countDownLatch.countDown()
+            }
+        }
+        countDownLatch.await()
+
+        // Verify alternative routes events were triggered.
+        assertNotNull(firstAlternative.routeProgress)
+        assertNotNull(firstAlternative.alternatives)
+
+        // Verify next alternative routes events were triggered without set routes
+        assertNotNull(nextAlternativeObserver.alternatives)
+        assertNotNull(nextAlternativeObserver.routeProgress)
+        // verify that set routes with  alternatives + 1 external alternatives is not triggered
```

Comment:
Why not?
`assertEquals(setRoute, nextAlternative)` implies that the routes passed via `setRoutes` are the same as the ones passed to the alternative observer, doesn't it?

Reply:
sorry, I see that it's a problem with tests. Need to fix: routes mustn't be the same! I need to cheks why the test passes 🤔